website_ranking_procedure.sql

```diff
@@ -0,0 +1,91 @@
+TRUNCATE TABLE  egstats.player_kpm;
+
+INSERT INTO egstats.player_kpm (player_name, match_date, kpm, game_id, modified_on) 
+SELECT pl.name, date_format(fg.end_time,'%Y-%m-%d') as match_date, fs.wins/((fs.warbird_time + fs.javelin_time + fs.spider_time + fs.leviathan_time + fs.terrier_time + fs.weasal_time + fs.lancaster_time + fs.shark_time)/60) as KPM, fs.game_id, now()
+FROM eg.flag_stats fs, eg.flag_games fg, eg.player pl, eg.pub_stats ps
+where fg.start_time > (select max(end_time) from eg.reset)
+and fs.game_id = fg.id
+and fs.player_id = pl.id
+and ps.player_id = pl.id
+and ((fs.warbird_time + fs.javelin_time + fs.spider_time + fs.leviathan_time + fs.terrier_time + fs.weasal_time + fs.lancaster_time + fs.shark_time)/60) > 10
+and ps.losses > '50'
+order by fs.wins/((fs.warbird_time + fs.javelin_time + fs.spider_time + fs.leviathan_time + fs.terrier_time + fs.weasal_time + fs.lancaster_time + fs.shark_time)/60) desc;
+
+TRUNCATE TABLE egstats.player_kpm_agg;
+
+INSERT INTO egstats.player_kpm_agg (player_name, kpm_avg, modified_on)
+SELECT player_name, avg(kpm) as kpm_avg, now() FROM egstats.player_kpm
+group by player_name;
+
+SELECT 
+    @wins_median:=MAX(wins) / 2 AS Win_Median
+FROM
+    eg.pub_stats;
+
+SELECT 
+    @loss_median:=MAX(losses) / 2 AS Loss_Median
+FROM
+    eg.pub_stats;
+
+SELECT 
+    @kd_median:=((wins) / (losses)) / 2 AS KD_Median
+FROM
+    eg.pub_stats
+    where losses > '50'
+ORDER BY ((wins) / (losses)) / 2 DESC
+LIMIT 1;
+
+SELECT 
+    @points_median:=(flag_points + kill_points) / 2 AS Points_Median
+FROM
+    eg.pub_stats
+WHERE losses > '50'
+ORDER BY (flag_points + kill_points) / 2 DESC
+LIMIT 1;
+
+SELECT 
+    @kpm_median:=max(kpm_avg) /2 from egstats.player_kpm_agg;
+
+TRUNCATE TABLE  egstats.zone_medians;
+
+Insert into egstats.zone_medians (wins_median, losses_median, kdratio_median, kpm_median, points_median, modified_on)
+values (@wins_median,@loss_median,@kd_median,@kpm_median,@points_median, NOW());
+
+Drop table player_rank;
```

Comment:
This is potentially dangerous since this .sql should be able to run against the DB at any time